core/src/main/java/bisq/core/api/model/PaymentAccountForm.java

```diff
@@ -0,0 +1,216 @@
+import com.google.gson.GsonBuilder;
+
+import javax.inject.Singleton;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+import java.util.Map;
+
+import java.lang.reflect.Type;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static bisq.core.payment.payload.PaymentMethod.getPaymentMethodById;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.lang.System.getProperty;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+
+/**
+ * <p>
+ * An instance of this class can write new payment account forms (editable json files),
+ * and de-serialize edited json files into {@link bisq.core.payment.PaymentAccount}
+ * instances.
+ * </p>
+ * <p>
+ * Example use case: (1) ask for a blank Hal Cash account form, (2) edit it, (3) derive a
+ * {@link bisq.core.payment.HalCashAccount} instance from the edited json file.
+ * </p>
+ * <br>
+ * <p>
+ * (1) Ask for a hal cash account form:  Pass a {@link bisq.core.payment.payload.PaymentMethod#HAL_CASH_ID}
+ * to {@link bisq.core.api.model.PaymentAccountForm#getPaymentAccountForm(String)} to
+ * get the json Hal Cash payment account form:
+ * <pre>
+ * {
+ *   "_COMMENT_": "Please do not edit the paymentMethodId field.",
+ *   "paymentMethodId": "HAL_CASH",
+ *   "accountName": "Your accountname",
+ *   "mobileNr": "Your mobilenr"
+ * }
+ * </pre>
+ * </p>
+ * <p>
+ * (2) Save the Hal Cash payment account form to disk, and edit it:
+ * <pre>
+ * {
+ *   "_COMMENT_": "Please do not edit the paymentMethodId field.",
+ *   "paymentMethodId": "HAL_CASH",
+ *   "accountName": "Hal Cash Acct",
+ *   "mobileNr": "798 123 456"
+ * }
+ * </pre>
+ * </p>
+ * (3) De-serialize the edited json account form:  Pass the edited json file to
+ * {@link bisq.core.api.model.PaymentAccountForm#toPaymentAccount(File)},
+ * and get a {@link bisq.core.payment.HalCashAccount} instance.
+ * <pre>
+ * PaymentAccount(
+ * paymentMethod=PaymentMethod(id=HAL_CASH,
+ *                             maxTradePeriod=86400000,
+ *                             maxTradeLimit=50000000),
+ * id=e33c9d94-1a1a-43fd-aa11-fcaacbb46100,
+ * creationDate=Mon Nov 16 12:26:43 BRST 2020,
+ * paymentAccountPayload=HalCashAccountPayload(mobileNr=798 123 456),
+ * accountName=Hal Cash Acct,
+ * tradeCurrencies=[FiatCurrency(currency=EUR)],
+ * selectedTradeCurrency=FiatCurrency(currency=EUR)
+ * )
+ * </pre>
+ */
+@Singleton
+@Slf4j
+public class PaymentAccountForm {
+
+    private final GsonBuilder gsonBuilder = new GsonBuilder()
+            .setPrettyPrinting()
+            .serializeNulls();
+
+    // Names of PaymentAccount fields to exclude from json forms.
+    private final String[] excludedFields = new String[]{
```

Comment:
Why are all these fields excluded? Some probably should be but at least SALT would be necessary to set manually to be able to recover an account.

Reply:
These are fields the user would not see or fill while creating a payment account in the UI.

But SALT shouldn't be excluded.  I'll fix that.

Reply:
Wouldn't it be better to have a `recoverpaymentacct salt` api method?

Do you or can you `createpaymentacct` with a predefined salt?

Reply:
You can set the salt, it's used as part of the hash to create the accountAgeWitness so it needs to be set to the same salt as previously if an account is to be created on a new client with the same details and account age preserved. Making it optional is reasonable I think. Default to generate a random salt, but if set then the user set salt will be used.

Reply:
Fixed in commit 9b4bdfc5ad8ff3e76296cb134b55ceba4397d158